<?xml version="1.0" encoding="UTF-8"?>
<launch>

<arg name="rviz"     default="true"/>
 <arg name="public_sim"   default="false"/>
 <arg name="sim_sufix" value="_public_sim"     if="$(arg public_sim)"/>
  <arg name="sim_sufix" value=""                unless="$(arg public_sim)"/>

 <node name="rviz" pkg="rviz" type="rviz" if="$(arg rviz)"
          args="-d $(find tiago_2dnav)/config/rviz/navigation_public_sim.rviz"/>


  <!-- marker detector -->
  <!--node pkg="perception_package" type="draw_boundingbox"  name="draw_boundingbox"/-->
  <!--node pkg="perception_package" type="from2dto3d" name="from2dto3d"/-->
  <node pkg="perception" type="object_detection" name="object_detection"/>

  <!-- Load definition of pregrasp motion into play_motion namespace -->
  <rosparam command="load" file="$(find manipulation)/config/pick_motions.yaml" /> 

  <!-- Pick & place server -->
  <node name="pick_and_place_server" pkg="manipulation" type="pick_and_place_server.py" output="screen">
      <rosparam command="load" file="$(find manipulation)/config/pick_and_place_params.yaml" />
      <param name="object_width"  value="0.01" />
      <param name="object_height" value="0.06" />
      <param name="object_depth"  value="0.06" />
  </node>

  <!-- Node exposing service to start looking for the object and trigger the picking -->
  <node name="pick_client" pkg="manipulation" type="pick_client.py" output="screen"/>  

  <group if="$(arg rviz)">
      <node name="pick_demo_rviz" pkg="rviz" type="rviz" args="-d $(find manipulation)/config/rviz/tiago_pick_demo.rviz" />
  </group>

</launch>
